File: iOS/TUILiveKit/Sources/Component/Gift/Model/TUIGiftData.swift
//
//  TUIGiftData.swift
//  TUILiveKit
//
//  Created by adamsfliu on 2025/6/16.
//

import Foundation
import RTCRoomEngine
import TUICore

public class TUIGiftData {
    public var giftCount: Int = 0
    public var giftInfo: TUIGiftInfo = TUIGiftInfo()
    public var sender: TUIUserInfo = TUIUserInfo()
    
    public var isAdvanced: Bool {
        giftInfo.resourceUrl.count > 0
    }
    
    public init() {}
    
    public init(_ giftCount: Int, giftInfo: TUIGiftInfo,  sender: TUIUserInfo) {
        self.giftCount = giftCount
        self.giftInfo = giftInfo
        self.sender = sender
    }
}

extension TUIUserInfo {
    var isSelf: Bool {
        userId == TUILogin.getUserID()
    }
}

File: Flutter/live_uikit_gift/lib/state/tui_gift_data.dart
import 'package:rtc_room_engine/rtc_room_engine.dart';

class TUIGiftData {
  int giftCount = 0;
  TUIGiftInfo giftInfo = TUIGiftInfo.fromJson({});
  TUIUserInfo sender = TUIUserInfo(
      userId: '', userName: '', avatarUrl: '', userRole: TUIRole.generalUser);

  TUIGiftData({int? giftCount, TUIGiftInfo? giftInfo, TUIUserInfo? sender}) {
    if (giftCount != null) {
      this.giftCount = giftCount;
    }
    if (giftInfo != null) {
      this.giftInfo = giftInfo;
    }
    if (sender != null) {
      this.sender = sender;
    }
  }

  @override
  String toString() {
    return 'TUIGiftData{giftCount: $giftCount, giftInfo: $giftInfo, sender: $sender}';
  }
}